robot/basestation/static/js/elements/navigation/navigation.js

```diff
@@ -1,191 +1,545 @@
+    let rover_position_listener = new ROSLIB.Topic({
+        ros: ros,
+        name: 'rover_position',
+        messageType: 'geometry_msgs/Point'
+    })
+    rover_position_listener.subscribe(function(message) {
+        $('#rover-latitude').text(message.x)
+        $('#rover-longitude').text(message.y)
+        $('#rover-heading').text(message.z)
+    })
+    // setup a subscriber for the antenna_goal topic
+    let antenna_goal_listener = new ROSLIB.Topic({
+        name: 'antenna_goal',
+        messageType: 'geometry_msgs/Point',
+        ros: ros
+    })
+    antenna_goal_listener.subscribe(function(message) {
+        $('#antenna-stats-rec-angle').text(parseFloat(message.x).toFixed(3))
+        $('#antenna-stats-dist-to-rover').text(parseFloat(message.y).toFixed(2))
+    })
+    // setup topics to communicate with GoalsNode
+    let create_goal_publisher = new ROSLIB.Topic({
+        ros: ros,
+        name: 'create_goal',
+        messageType: 'mcu_control/RoverGoal'
+    })
+    let delete_goal_publisher = new ROSLIB.Topic({
+        ros: ros,
+        name: 'delete_goal',
+        messageType: 'std_msgs/String'
+    })
+    goal_list_subscriber.subscribe(function(message) {
+        goalList = message.goal_list
+        if (goalCount != goalList.length) {
+            goalCount = goalList.length
+
+        }
+        if (goalCount > 0) {
+            goal_latitude.set(goalList[0].lat)
+            goal_longitude.set(goalList[0].long)
+            $('#goal-stats-lat').text(goalList[0].lat.toFixed(6))
+            $('#goal-stats-long').text(goalList[0].long.toFixed(6))
+        } else {
+            goal_latitude.set(false)
+            goal_longitude.set(false)
+            $('#goal-stats-lat').text('----')
+            $('#goal-stats-long').text('----')
+        }
+
+        // this is a meme implementation of a loop to update the params without resorting to a message type + subscriber + publisher
+        updateAntennaParams()
+    })
+    // setup gps parameters for antenna directing
+    let antenna_latitude = new ROSLIB.Param({
+        ros: ros,
+        name: 'antenna_latitude'
+    })
+    let antenna_longitude = new ROSLIB.Param({
+        ros: ros,
+        name: 'antenna_longitude'
+    })
+    let antenna_start_dir = new ROSLIB.Param({
+        ros: ros,
+        name: 'antenna_start_dir'
+    })
+    let has_gps_goal = new ROSLIB.Param({
+        ros: ros,
+        name: 'has_gps_goal'
+    })
+    // setup gps parameters for rover goals
+    let goal_latitude = new ROSLIB.Param({
+        ros: ros,
+        name: 'goal_latitude'
+    })
+    let goal_longitude = new ROSLIB.Param({
+        ros: ros,
+        name: 'goal_longitude'
+    })
+    // setup a subscriber for the rover_goal topic
+    new ROSLIB.Topic({
+        ros: ros,
+        name: 'rover_goal',
+        messageType: 'geometry_msgs/Point'
+    });
+
+    goalList = []
+    goalCount = 0;
+
+    initNavigationPanel()
+
+    function initNavigationPanel() {
```

Comment:
I'd like to suggest to ignore the params, under the assumption that we are going to remove the use of ros params for the initial data seeds and simply stick to publishing this information to the according topics. @vashmata, @MartensCedric,  makes sense?

Reply:
Hmm shoot, I've been using ros params to ensure that the goal list and antenna data are kept between page reloads and are synced between operators. I can't think of any other way to ensure that this data would be shared across different multiple operators.